src/components/group-common.js

```diff
@@ -104,76 +106,103 @@ export const NewUserModal = ajaxCaller(class NewUserModal extends Component {
+            this.setState({ submitting: true })
+            await inviteFunction(userEmail)
+            await this.submit()
           }
-        },
-        suggestions,
-        style: { fontSize: 16 },
-        theme: { suggestion: { padding: 0 } }
-      }),
-      h(FormLabel, ['Role']),
-      h(LabeledCheckbox, {
-        checked: isAdmin,
-        onChange: () => this.setState({ roles: [isAdmin ? userLabel : adminLabel] })
+        }, ['Yes']),
+        cancelText: 'No',
+        onDismiss: () => this.setState({ confirmAddUser: false })
       }, [
-        label({ style: { margin: '0 2rem 0 0.25rem' } }, [`Can manage users (${adminLabel})`])
-      ]),
-      footer && div({ style: { marginTop: '1rem' } }, [footer]),
-      submitError && div({ style: { marginTop: '0.5rem', textAlign: 'right', color: colors.danger() } }, [submitError]),
-      submitting && spinnerOverlay
-    ])
+        'Add ', b(userEmail), ' to the group anyway?',
+        submitting && spinnerOverlay
+      ])],
+      h(Modal, {
+        onDismiss,
+        title,
+        okButton: h(ButtonPrimary, {
+          tooltip: Utils.summarizeErrors(errors),
+          onClick: async () => await this.checkUserRegistration() ? this.submit() : this.setState({ confirmAddUser: true }),
+          disabled: errors
+        }, ['Add User'])
+      }, [
+        h(RequiredFormLabel, ['User email']),
+        h(AutocompleteSearch, {
+          autoFocus: true,
+          value: userEmail,
+          onChange: v => this.setState({ userEmail: v }),
+          renderSuggestion: suggestion => div({ style: styles.suggestionContainer }, [
+            div({ style: { flex: 1 } }, [
+              !canAdd(suggestion) && h(TooltipTrigger, {
+                content: 'Not a valid email address'
+              }, [
+                icon('warning-standard', { style: { color: colors.danger(), marginRight: '0.5rem' } })
+              ]),
+              suggestion
+            ])
+          ]),
+          onSuggestionSelected: selection => {
+            this.setState({ userEmail: selection })
+          },
+          onKeyDown: e => {
+            // 27 = Escape
+            if (e.which === 27 && !!userEmail) {
+              this.setState({ userEmail: '' })
+              e.stopPropagation()
+            }
+          },
+          suggestions,
+          style: { fontSize: 16 },
+          theme: { suggestion: { padding: 0 } }
+        }),
+        h(FormLabel, ['Role']),
+        h(LabeledCheckbox, {
+          checked: isAdmin,
+          onChange: () => this.setState({ roles: [isAdmin ? userLabel : adminLabel] })
+        }, [
+          label({ style: { margin: '0 2rem 0 0.25rem' } }, [`Can manage users (${adminLabel})`])
+        ]),
+        footer && div({ style: { marginTop: '1rem' } }, [footer]),
+        submitError && div({ style: { marginTop: '0.5rem', textAlign: 'right', color: colors.danger() } }, [submitError]),
+        (checkingUser || submitting) && spinnerOverlay
```

Comment:
nit: Since these two states are mutually exclusive, we might be able to simplify the flow a bit by using a unified `busy` flag.